app/pb/account.proto

```diff
@@ -77,3 +98,49 @@ message GetContributorFormInfoRes {
 message MarkContributorFormFilledReq {
   bool filled_contributor_form = 1;
 }
+
+message ChangePhoneReq {
+  // New phone number in E.164 format, or empty string to remove phone number.
+  // The phone number will be visible to moderators in your area, verified or not.
+  string phone = 1;
+
+  // If you have a previously verified number, changing your number and losing
+  // the verified status will fail if false is passed.
+  bool remove_verification = 2;
+}
+
+message SendVerificationSMSReq {
+}
+
+message SendVerificationSMSRes {
+  enum SendVerificationSMSStatus {
+    SMS_RESULT_CODE_SENT = 0;
+    SMS_RESULT_ALREADY_VERIFIED = 1;  // No code sent since the same number is already verified, and a renewal can't be done yet
+    SMS_RESULT_RATELIMIT = 2;   // Hold your horses, contact admin to send more than one SMS in a short period
+    SMS_RESULT_UNSUPPORTED_OPERATOR = 3;  // Sorry, we don't know how to send SMS to your number
+    SMS_RESULT_OPERATOR_ERROR = 4;  // Unknown, see operator_error_message.
+    SMS_RESULT_CONSENT_NEEDED = 5;  // Accept a later ToS version to allow couchers to use external services for sms delivery
+    SMS_RESULT_NO_PHONE_NUMBER = 6;  // User hasn't inputed a phone number yet
+  }
+  SendVerificationSMSStatus status = 1;
+
+  // Empty string in all cases except when status == SMS_RESULT_OPERATOR_ERROR
+  // Might not be english.
+  string operator_error_message = 2;
+}
+
+message VerifyPhoneReq {
+  // Should be a numerical string with Luhn algorithm mod 10 checksum equals 0.
+  // Frontends could do that pre-check.
+  string token = 1 [(sensitive) = true];
```

Comment:
Hmm, why is a checksum needed? When I get SMS codes they're these days 6 digits mostly; I think it'd be good to keep them short at the expense of not having a checksum digit.

Reply:
I thought it would be fun to have. Frontends are free to ignore that though. 5 digits + checksum with three attempts still gives you only 0.003 % chance of guessing it.

Let's hardcode six digits in the interface.